Update docs on new model couplers (#3723)
PecanProject/pecan: book_source/03_topical_pages/03_pecan_xml.Rmd

* `name` : (required) the name of the PFT, which must *exactly* match the name in the PEcAn database.
* `outdir`: (optional) Directory path in which PFT-specific output will be stored during meta-analysis and sensitivity analysis. If not specified (recommended), it will be written into `<outdir>/<pftname>`.
* `constants`: (optional) this section contains information that will be written directly into the model specific configuration files. For example, some models like ED2 use PFT numbers instead of names for PFTs, and those numbers can be specified here. See documentation for model-specific code for details.

> * `posterior.files` (Optional if database is available, required if not) this tag helps to signal write.config functions to use specific posterior/prior files (such as HPDA or MA analysis) for generating samples without needing to access to the bety database.

Reviewer: I'd rephrase as "(Required unless database is available)" so that Required is the first word the reader sees, rather than that important info being buried

Author: ```suggestion
* `posterior.files` (Required unless database is available) this tag helps to signal write.config functions to use specific posterior/prior files (such as HPDA or MA analysis) for generating samples without needing to access to the bety database.
```
